src/emissions.py

```diff
@@ -0,0 +1,92 @@
+# emissions = pd.read_csv('data/EC_Factors_Passenger.csv')
+# with open("static\TestRoute_Wag_Roest.geojson") as gj_file:
+#  gj = geojson.load(gj_file)
+# speeds = []
+# for feature in gj['features']:
+#   print(feature['attributes'])
+
+# route =
+# speed = sample.geojson[]
+# print(emissions.head())
+
+# final formula (Alpha * Speed ^ 2 + Beta * Speed + Gamma + (Delta/Speed) ) / (Epsilon * Speed^2 + Zita * Speed + Hta)
+
+# input will be speeds for each segment in csv
+# path is a path to the file
+# def constructSpeedDF (path):
+# spDF = read_csv(str(path))
+
+# create a sample dataFrame with speeds per segment
+
+speedPerSegmentDict = {'segment': [0, 1, 2, 3, 4],
+                       'Speed': [9, 70, 55, 80, 70]}
+
+speedPerSegmentDF = pd.DataFrame(speedPerSegmentDict)
+speedPerSegmentDF.to_csv('data/b_SpeedFlow_sample.csv')
+
+# this function derive values for further emission factors calculation
+def derive_values_for_calc(path_input_emis_calc, category, fuel, segment, fuel_st, technology, pollutant):
+    # maybe we should check that all the inputs are strings before make df
+    values_for_emis_calc = pd.read_csv(path_input_emis_calc)
+    df_values = values_for_emis_calc[(values_for_emis_calc['Category'] == category) &
+                                     (values_for_emis_calc['Fuel'] == fuel) &
+                                     (values_for_emis_calc['Segment'] == segment) &
+                                     (values_for_emis_calc['Euro Standard'] == fuel_st) &
+                                     (values_for_emis_calc['Technology'] == technology) &
+                                     (values_for_emis_calc['Pollutant'] == pollutant)]
+    a = df_values['Alpha']
+    b = df_values['Beta']
+    g = df_values['Gamma']
+    d = df_values['Delta']
+    e = df_values['Epsilon']
+    z = df_values['Zita']
+    h = df_values['Hta']
+    return a, b, g, d, e, z, h
+
+
+# This a separate function for formula
+def formula_emission_factor(a, b, g, d, e, z, h, speed):
+    emis_f = a * speed ** 2 + b * speed + g + (d / speed) / (e * speed ** 2 + z * speed + h)
+    return emis_f
+
+
+# Unite two previous functions into one, where output is the dataframe with emission factor per segment
+def calculate_emission_factor(path_input_emis_calc, category, fuel, segment, fuel_st, technology, pollutant,
+                              path_speeds_per_seg):
+    a, b, g, d, e, z, h = derive_values_for_calc(path_input_emis_calc, category, fuel, segment, fuel_st, technology,
+                                                 pollutant)
+    speed_per_seg = pd.read_csv(path_speeds_per_seg, index_col='segment')
+    speed_per_seg.drop(columns=['Unnamed: 0'])
+    emis_fs = []
+    for i in range(speed_per_seg.index.size):
+        speed = speed_per_seg.loc[i, 'Speed']
+        if speed < 10.00:
+            speed = 10.00
+            emis_f = formula_emission_factor(a, b, g, d, e, z, h, speed)
```

Comment:
You're now repeating the line `emis_f = formula_emission_factor(a, b, g, d, e, z, h, speed)` three times, you can also run it only after the if statement. 
Also, it is probably better to use a different variable name in lines 74, 77 and 80; because `speed` is already used

Reply:
Yes, I was just a bit confused with the scope yesterday, so I've decided to put it here for now. I'll try to put after 'if' then.

The name is the same because I need to calculate the emission factor for speed < 10 as if the speed is 10, so it's the same variable and that's why the same name.